pkg/tfvars/openstack/openstack.go

```diff
@@ -104,10 +104,26 @@ func TFVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, cloud string, exter
 			return nil, errors.Errorf("Unsupported URL scheme: '%v'", url.Scheme)
 		}
 
-		err = uploadBaseImage(cloud, localFilePath, imageName, infraID)
+		useImageImport, err := isImageImportSupported(cloud)
 		if err != nil {
 			return nil, err
 		}
+
+		err = uploadBaseImage(cloud, localFilePath, imageName, infraID, useImageImport)
+		if err != nil {
+			// If we tried to upload the data with the image import mechanism and failed,
+			// let's disable it and try legacy uploading.
+			if useImageImport {
+
+				err = uploadBaseImage(cloud, localFilePath, imageName, infraID, false)
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				return nil, err
```

Comment:
we loose the first error. i.e. if the first time it fails but it was due to something not related to useImageImport=True, then we'll loose the first error. It could make it hard to debug. any way to annotate the first error the second. Or at least log it?